lib/streamlit/caching.py

```diff
@@ -442,6 +452,10 @@ def cache(
         Suppress warnings about calling Streamlit functions from within
         the cached function.
 
+    hash_funcs : dict or None
+        Mapping of types to functions for objects that we can't hash. The function should
+        accept an object as a parameter.
```

Comment:
This doc needs more love: 

> Mapping of types to hash functions. This used to override the behavior of the hasher inside of the caching mechanism: when the hasher encounters an object, it will first check to see if the type matches a key in this dict and, if so, will use the provided function to generate a hash for it. See below for an example of how this can be used.

...and then add an example in the `Examples` section.